components/brave_ads/core/internal/creatives/creative_ad_util.h

```diff
@@ -30,6 +33,27 @@ size_t TargetedCreativeAdCount(const T& creative_ads) {
       });
 }
 
+template <typename T>
+T DeduplicateCreativeAds(const T& creative_ads) {
+  base::flat_set<std::string> uuids;
```

Comment:
How many times we have here?
in flat_set insert() takes O(N) time, so it N >= 100 this could be an issue.

Reply:
PR is in draft. This was a quick fix but working on a better fix. I will move PR out of draft when ready. Thanks